rebuild/src/public_key.h

```diff
@@ -0,0 +1,96 @@
+#ifndef BLST_TS_PUBLIC_KEY_H__
+#define BLST_TS_PUBLIC_KEY_H__
+
+#include <memory>
+
+#include "addon.h"
+#include "blst.hpp"
+#include "napi.h"
+
+#define BLST_TS_PUBLIC_KEY_LENGTH_COMPRESSED 48U
+#define BLST_TS_PUBLIC_KEY_LENGTH_UNCOMPRESSED 96U
+
+typedef struct {
+    std::unique_ptr<blst::P1_Affine> smart_pointer;
+    blst::P1_Affine *raw_point;
+} P1AffineGroup;
+
+class P1Wrapper {
+   public:
+    virtual ~P1Wrapper() = default;
+    virtual bool IsInfinite() = 0;
+    virtual bool InGroup() = 0;
+    virtual void Serialize(bool compress, blst::byte *out) = 0;
+    virtual void AddTo(blst::P1 &point) = 0;
+    virtual blst::P1 MultiplyBy(
+        blst::byte *rand_bytes, size_t rand_bytes_length) = 0;
+    virtual P1AffineGroup AsAffine() = 0;
+};
+
+class P1 : public P1Wrapper {
```

Comment:
Same as noted elsewhere, `final` might improve correctness and performance.

Reply:
I'll omit complaining about `final` elsewhere. If you feel it makes sense to use that keyword, please check other derived classes for potential usage.